common/infrastructure/src/main/kotlin/no/nav/su/se/bakover/common/infrastructure/job/StoppableJob.kt

```diff
@@ -0,0 +1,122 @@
+package no.nav.su.se.bakover.common.infrastructure.job
+
+import arrow.core.Either
+import no.nav.su.se.bakover.common.CorrelationId
+import no.nav.su.se.bakover.common.infrastructure.correlation.withCorrelationId
+import no.nav.su.se.bakover.common.sikkerLogg
+import org.jetbrains.kotlin.utils.addToStdlib.ifTrue
+import org.slf4j.Logger
+import java.time.Duration
+import java.util.Date
+import java.util.Timer
+import java.util.TimerTask
+import kotlin.concurrent.fixedRateTimer
+
+interface StoppableJob {
+    val jobName: String
+
+    /**
+     * Idempotent. Forventer bare at ingen nye jobber blir startet. Pågående kjører ferdig.
+     */
+    fun stop()
+}
+
+/**
+ * Starter en jobb som kjører med fast intervall.
+ *
+ * @param job Wrappes i en correlationId og en try-catch for å logge eventuelle feil.
+ * @param runJobCheck Liste av RunJobCheck som må returnere true for at jobben skal kjøre. Ved tom liste, kjøres jobben alltid.
+ */
+fun startStoppableJob(
+    jobName: String,
+    initialDelay: Duration,
+    intervall: Duration,
+    log: Logger,
+    runJobCheck: List<RunJobCheck>,
+    job: (CorrelationId) -> Unit,
+): StoppableJob {
+    log.info("Starter skeduleringsjobb '$jobName'. Intervall: hvert ${intervall.toMinutes()}. minutt. Initial delay: ${initialDelay.toMinutes()} minutt(er)")
+    return startStoppableJob(
+        jobName = jobName,
+        log = log,
+        runJobCheck = runJobCheck,
+        job = job,
+    ) {
+        fixedRateTimer(
+            name = jobName,
+            daemon = true,
+            initialDelay = initialDelay.toMillis(),
+            period = intervall.toMillis(),
+            action = it,
+        )
+    }
+}
+
+/**
+ * Starter en jobb som kjører med fast intervall.
+ *
+ * @param job Wrappes i en correlationId og en try-catch for å logge eventuelle feil.
+ * @param runJobCheck Liste av RunJobCheck som må returnere true for at jobben skal kjøre. Default er en tom liste. Ved tom liste, kjøres jobben alltid.
+ */
+fun startStoppableJob(
+    jobName: String,
+    startAt: Date,
+    intervall: Duration,
+    log: Logger,
+    runJobCheck: List<RunJobCheck>,
+    job: (CorrelationId) -> Unit,
+): StoppableJob {
```

Comment:
kan være fint å få frem at denne vil starte på et gitt tidspunkt - sammenlignet med den over. måtte lese nøyere for å få med meg det